### DISCUSSION TITLE
uql: using grafana global variables

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#905] [Q&A] [answered]
Greetings,

Is is possible to capture the grafana `__from` variable as follows?

```
parse-csv
| extend "datetime"=strcat('2024-07-03 ', "time"), "grafana_start_date"=${__from:date:YYYY-MM-DD}
```

Thanks!

**Comment by canob** [comment]
Hi @yosiasz ,

Do you know if it is even possible to use a Grafana variable on UQL?

Because I'm trying this:

```
parse-json
| project "resources"
parse-json
| extend "IOC_type"="type", "IOC_value"="value", "timestamp"=todatetime("created_on"), "customer"=${customer}, "device"=strcat('Crowdstrike')
| project "IOC_type", "IOC_value", "timestamp", "customer", "device"
```

And I'm getting "No data" (it is working if I delete "customer" field, that is, not using the variable).
I tried with all the options here: https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/ ($customer, ${customer}, etc.) with no luck.

Thanks

**Reply by yesoreyeram** [reply]
if the customer field is string, you may need to wrap with singlequote around the variable. Example

```kusto
| extend "customer"='${customer}'
```

**Reply by canob** [reply]
Hi @yesoreyeram,

`"customer"='${customer}'` was one of the many options that I tried, with no luck. Even that the variable is a text (it is a simple custom type variable, with two values, to test), I tried with `"customer"='${customer:text}'` with no luck too, always "No data".

The only option with which I get all the other fields, but the `customer` field empty, is using `"customer"=${customer:doublequote}`, if it helps you to troubleshoot

```
parse-json
| project "resources"
parse-json
| extend "IOC_type"="type", "IOC_value"="value", "timestamp"=todatetime("created_on"), "customer"=${customer:doublequote}, "device"=strcat('Crowdstrike')
| project "IOC_type", "IOC_value", "timestamp", "customer", "device"
```

![image](https://github.com/user-attachments/assets/81407cdb-467a-4806-b9f8-e4f18f29c9e8)

**Reply by yesoreyeram** [reply]
see if the docs https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/filters/ helps

**Reply by canob** [reply]
Finally, the one that worked:

`| extend "customer"=strcat('$customer')`

**Comment by yesoreyeram** [comment, marked as answer]
@yosiasz - yes infinity supports the time macros. better to use time macros as specified in https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/macros/#time-macros-__timefrom-and-__timeto 

> note: UQL doesn't support these macros in extend/project commands. But that should work with JSONata. [Example](https://play.grafana.org/goto/HUBVKP_SR?orgId=1). This time macro should work in other places such as root selector in JSON backend.
